liferay-content-management/liferay-portal
# LPD-18231 Remove trailing slashes from AM paths

# What is this trying to solve?

https://liferay.atlassian.net/browse/LPD-18231

Adaptive Media store paths contain trailing slashes. This has no effect in most store implementations, but in GCS this is significant.

# How am I fixing it?

Remove trailing slashes from Adaptive Media paths; also, make GCS store more lenient with leading/trailing slashes to avoid similar problems in the future.

# How can you verify that it works?

This PR includes unit tests that verify the low level behavior. You may also follow the steps described in https://liferay.atlassian.net/browse/LPD-18231.
